design/authenticators/authn_k8s/inject_client_cert_solution_design.md

```diff
@@ -0,0 +1,255 @@
+[//]: # "You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#"
+
+| **Scenario**                                             | **Log message**                                              |
+| -------------------------------------------------------- | ------------------------------------------------------------ |
+| Timeout reached for injection kubectl exec command       | Command timed out after {0} seconds in container '{1}' of pod '{2}' |
+| Before copying the file to the container                 | Copying file to {0-container-name}:{1-cert-file-path} in {2-pod-namespace}/{3-pod-name}. stdout and stderr will be written to {4-log-file-path}" |
+| After starting the cert injection and closing the socket | Initialized SSL certificate copy successfully                |
+
+Additionally, we should address [this issue](https://github.com/cyberark/conjur/issues/1840) as part of this effort.
+
+### Audit 
+
+[//]: # "Does this solution require additional audit messages?"
+There are no new audit messages in this fix. We already write the inject-client-cert request to the audit log.
+
+## Documentation
+[//]: # "Add notes on what should be documented in this solution. Elaborate on where this should be documented. If the change is in open-source projects, we may need to update the docs in github too. If it's in Conjur and/or DAP mention which products are affected by it"
+This is an internal process so it doesn't require any new documentation. A CHANGELOG entry will be added about the response code change in "inject-client-cert"
+
+## Delivery plan
+[//]: # "Break the solution into tasks"
+
+### Total
+[//]: # "Write the total estimation"
+
+EE: 11-16 days
+
+### Parallelism
+
+[//]: # "Suggest how many team members can work on the solution in parallel"
+
+- 2 people can work on this solution but it's ok to have only 1.
+
+### Spikes and researches
+
+[//]: # "Note and estimate the tasks which are related to spikes and researches which needed for the solution"
+
+This spike will be done only if we decide to go with option 1 instead of option 4.
+
+- What happens if we send an "ls" command on a file that is being populated by a "tar" command? Will we get a response that the file exist although it is not fully populated?
+  - EE:  1 day
+  
+### Implementation 
+[//]: # "Note and estimate the tasks which are related to the solution's code implementation"
+
+- Implement solution according to the design
+  - EE: 5-10 days (depending on solution - 4 will be faster)
+- Fix issue - [Kubectl exec Websocket logs do not have tags](https://github.com/cyberark/conjur/issues/1840)
+  - EE: 2 days
+- Print error to authn-client log
+  - Do we want to delete the log file after we log its content?
+  - EE: 2 days
+
+### Testing 
+
+[//]: # "Note and estimate the tasks which are related to writing/performing tests [automation, manual tests, etc.]"
+
+- Write tests according to the test plan
+  - EE: 4 days
```

Comment:
do you have a clear sense of the tests you'll be adding? are there unit/rspec tests we can add, or are you planning primarily to add cucumber or e2e tests?

Reply:
i'm not adding any integration tests, just fixing them to verify 202 instead of 200.
I am not sure that we need any new UTs but i'll check that. it is part of the 4 days.